api/v1alpha1/gateway_types.go

```diff
@@ -56,22 +56,61 @@ type GatewaySpec struct {
 	// Listeners associated with this Gateway. Listeners define what addresses,
 	// ports, protocols are bound on this Gateway.
 	Listeners []Listener `json:"listeners"`
-	// Routes associated with this Gateway. Routes define protocol-specific
-	// routing to backends (e.g. Services).  Typically the resource is
-	// "httproute" or "tcproute" in group "networking.x.k8s.io", or an
-	// implementation may support other resources.
-	Routes []RouteObjectReference `json:"routes"`
+
+	// Servers binds virtual servers that are hosted by the gateway
+	// to listeners.
+	//
+	// An implementation must validate that Dedicated listeners have no
+	// more than 1 server binding.
+	//
+	// An implementation must validate that all the servers bound to the
+	// same combined listener have a compatible discriminator.
+	// TODO(jpeach): describe consequences of this validating failing.
+	Servers []ServerBinding `json:"servers"`
 }
 
-const (
-	// HTTPProcotol constant.
-	HTTPProcotol = "HTTP"
-	// HTTPSProcotol constant.
-	HTTPSProcotol = "HTTPS"
-)
+// ServerBinding specifies which listener a virtual server should
+// be attached to. A server can be bound to 1 or more listeners.
+type ServerBinding struct {
+	// ListenerName is the unique name of a listener.
+	//
+	// TODO(jpeach): We could let an empty name mean all listeners, but
+	// there are trade-offs to that. For now this field must not be empty.
+	//
+	// +required
+	ListenerName `json:"listenerName"`
+	// Server is a reference to the virtual server to be attached.
+	//
+	// +required
+	Server ServerObjectReference `json:"server"`
+}
+
+type ListenerProtocolType string
+
+const ListenerProtocolTCP ListenerProtocolType = "TCP"
+const ListenerProtocolUDP ListenerProtocolType = "UDP"
+
+type Listenertype string
+
+// DedicatedListenerType describes a listener with exactly on attached server.
+// All traffic received by this listener is terminated by this server.
+const DedicatedListenerType ListenerType = "Dedicated"
+
+// CombinedListenerType describes a listener what may have multiple attached
+// servers. The protocol recognized by this listener must have a discriminator
+// that can be used to select the appropriate server for the incoming request.
+// In the case of a streaming data over TLS, the ALPN protocol name can sepect
```

Comment:
sepect... do you mean inspect?

Reply:
I think I meant to say "the ALPN protocol name can be used to select".